faker/providers/address/__init__.py

```diff
@@ -82,3 +82,40 @@ def country_code(self, representation=ALPHA_2):
             return self.random_element(self.alpha_3_country_codes)
         else:
             raise ValueError("`representation` must be one of `alpha-2` or `alpha-3`.")
+
+    def administrative_unit(self):
+        for name in ["province", "region", "state", "prefecture", "county"]:
+            method = getattr(self, name, None)
+            if method:
+                return method()
+        msg = f"No state or other high level administrative unit found for this locale: {self.__class__.__module__}"
+        raise AttributeError(msg)
+
+    def current_country_code(self):
+        try:
+            return self.__lang__.split("_")[1]
+        except IndexError:
+            raise AttributeError("Country code cannot be determmined from locale")
+
+    def current_country(self):
+        try:
+            current_country_code = self.current_country_code()
+        except AttributeError:
+            return "No Country"
+        current_country = [tz['name']
+                           for tz in date_time.Provider.countries
+                           if tz['alpha-2-code'] == current_country_code]
+        if len(current_country) == 1:
+            return current_country[0]
+        elif len(current_country) > 1:
+            raise ValueError(f"Ambiguous country for country code {current_country_code}: {current_country}")
+        elif current_country_code == "AA":
+            return "No Country"
+        elif current_country_code == "PS":
```

Comment:
`PS` should work without special-case. Every provider should have it. Where is it failing?

Reply:
Palestine does not have a time-zone entry in date_time.countries. Perhaps the time-zone people are not as open to Palestinian independence as the 2-letter ISO people. Same for Taiwan.

Reply:
I see 2 timezone in the [TZ Database](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) for `PS`: `Asia/Gaza` and `Asia/Hebron`. We should probably add both?